### ISSUE TITLE
Which changes are non-notable?

### ISSUE THREAD

**Issue by jeremykohn** [olivierlacan/keep-a-changelog#421]
If a changelog is "a curated, chronologically ordered list of *notable changes*" (emphasis added), that implies some changes are *not* notable and therefore should not be included. 

So, two questions:

### (1) What types of changes to a coding project are non-notable?

Some possibilities:

Generally revising code to
- [ ] improve performance?
- [ ] improve maintainability?
- [ ] improve readability?

Revising comments
- [ ] in a substantive way?
- [ ] in a stylistic way?

Adding, removing, or changing
- [ ] console.log or print statments?
- [ ] debug statments?

Changing (if not significant for code execution)
- [ ] variable names?
- [ ] indentation?
- [ ] whitespace?

Adding/removing empty or unused
- [ ] styles?
- [ ] function definitions?

Changing the sequence of
- [ ] array elements?
- [ ] function arguments?
- [ ] variable declarations or assignments?

Assuming none of those changes will significantly affect the code's final output or end result.

And a related question:

### (2) If certain changes are not listed in a changelog, where should they be listed (if anywhere)?

- [ ] Git commit logs only?
- [ ] Readme file?
- [ ] A separate file called "Edits," "Revisions," "Differences,"  etc?
- [ ] User-facing release notes? (Most likely not; if developers aren't interested in non-notable changes, then customers or end-users probably wouldn't be interested either)

**Comment by Archmonger** [comment]
My thoughts are it's kept open-ended for each project to decide what is or isn't notable. For example, fixing a typo in a comment is almost always non-notable.

But, increasing performance may or may not be notable, depending on how much of an improvement was made. I don't think `keep-a-changelog` should bound something like this. Maybe adding a tidbit in the FAQ saying "it's up for you to decide what is notable" is the most I would think fits the scope of this project.

**Comment by GitReddyHub** [comment]
I agree with @Archmonger, the criteria for what is a notable change and what is not shouldn't be set in stone. However, this criteria could be included as guidelines (so developers have an idea of when to add to the changelog). As for the second question that @jeremykohn posed, I'm not that experienced with managing changelogs, but I feel that just adding them to git commit logs would be fine, along with a "fixed minor issues" in the release notes if the developer really wants to.

**Comment by suriyawutkhongprom** [comment]
> If a changelog is "a curated, chronologically ordered list of *notable changes*" (emphasis added), that implies some changes are *not* notable and therefore should not be included. 
> 
> So, two questions:
> 
> ### (1) What types of changes to a coding project are non-notable?
> 
> Some possibilities:
> 
> Generally revising code to
> - [ ] improve performance?
> - [ ] improve maintainability?
> - [ ] improve readability?
> 
> Revising comments
> - [ ] in a substantive way?
> - [ ] in a stylistic way?
> 
> Adding, removing, or changing
> - [ ] console.log or print statments?
> - [ ] debug statments?
> 
> Changing (if not significant for code execution)
> - [ ] variable names?
> - [ ] indentation?
> - [ ] whitespace?
> 
> Adding/removing empty or unused
> - [ ] styles?
> - [ ] function definitions?
> 
> Changing the sequence of
> - [ ] array elements?
> - [ ] function arguments?
> - [ ] variable declarations or assignments?
> 
> Assuming none of those changes will significantly affect the code's final output or end result.
> 
> And a related question:
> 
> ### (2) If certain changes are not listed in a changelog, where should they be listed (if anywhere)?
> 
> - [ ] Git commit logs only?
> - [ ] Readme file?
> - [ ] A separate file called "Edits," "Revisions," "Differences,"  etc?
> - [ ] User-facing release notes? (Most likely not; if developers aren't interested in non-notable changes, then customers or end-users probably wouldn't be interested either)
